### ISSUE TITLE
Configuration Groups - Unable to assign template to single device only 

### ISSUE THREAD

**Issue by benkelly12** [CiscoDevNet/terraform-provider-sdwan#336]
As per title, there doesn't seem to be an option to assign a template to a single device only, this is required when you're using a dual router setup with TLOC extensions. This means I can't define the full configuration in Terraform

Also you don't appear to be able to assign device to configuration groups yet.

**Comment by danischm** [comment]
This is work in progress. The next release will include a resource to assign devices to config groups and deploy them.

**Comment by benkelly12** [comment]
> This is work in progress. The next release will include a resource to assign devices to config groups and deploy them.

Hi, Thank you 
What about the ability to have individual templates apply only to one device or the other, this is possible in the GUI but I couldn't find a way to do it. This is not done during the assignment of devices, this is part of the actual configuration group with edge01 and edge02 being logical representations of the devices you will add when you assign the template. 

An example of use case of this is a branch with two routers, one with an internet connection and one with MPLS with TLOC extension between them. There is a lot of config that would be unique to each device in that scenario. 

Screenshot of the GUI attached

Looks like you're on about a two month release cadence?

**Comment by danischm** [comment]
> What about the ability to have individual templates apply only to one device or the other, this is possible in the GUI but I couldn't find a way to do it. This is not done during the assignment of devices, this is part of the actual configuration group with edge01 and edge02 being logical representations of the devices you will add when you assign the template.

I believe this is what can be configured with `topology_devices` and `topology_site_devices` attributes of `sdwan_configuration_group` resource. From what I remember this looks a bit different compared to the UI, so capturing the API payloads while doing it in the UI might help.

**Comment by benkelly12** [comment]
> > What about the ability to have individual templates apply only to one device or the other, this is possible in the GUI but I couldn't find a way to do it. This is not done during the assignment of devices, this is part of the actual configuration group with edge01 and edge02 being logical representations of the devices you will add when you assign the template.
> 
> I believe this is what can be configured with `topology_devices` and `topology_site_devices` attributes of `sdwan_configuration_group` resource. From what I remember this looks a bit different compared to the UI, so capturing the API payloads while doing it in the UI might help.

You are right, it makes a post to the profile and then a put to the configuration-group (attached both). 
Having reviewed topology devices documentation though I don't think currently it will work. It seems to be what sets the logical device names within the configuration-group, but there doesn't appear to be the functionality to then link templates to specific devices.

**Comment by benkelly12** [comment]
> > > What about the ability to have individual templates apply only to one device or the other, this is possible in the GUI but I couldn't find a way to do it. This is not done during the assignment of devices, this is part of the actual configuration group with edge01 and edge02 being logical representations of the devices you will add when you assign the template.
> > 
> > 
> > I believe this is what can be configured with `topology_devices` and `topology_site_devices` attributes of `sdwan_configuration_group` resource. From what I remember this looks a bit different compared to the UI, so capturing the API payloads while doing it in the UI might help.
> 
> You are right, it makes a post to the profile and then a put to the configuration-group (attached both). Having reviewed topology devices documentation though I don't think currently it will work. It seems to be what sets the logical device names within the configuration-group, but there doesn't appear to be the functionality to then link templates to specific devices.

Think I've worked it out from the API call actually 
it appears to add the parcel ID as an unsupported type on the opposite device. So if I want to apply a profile to only EdgeDevice_01 it will add that profile as an unsupported type on Edgedevice_02

**Comment by tzarski0** [comment]
This is already addresses in v0.6.0.
